### ISSUE TITLE
Some questions when reproducing code

### ISSUE THREAD

**Issue by sunshinelium** [Yu-Fangxu/DARec#1]
hello, thanks for your open code. In my comparative experiment, I changed the evaluation criterion from RMASE to top-N recommendation. We selected the largest N items as recommendations according to the predicted values, and used HR and NDCG as measurement standards. However, I found that the effect was a little poor, and if you have tried this evaluation method. Or some possible reasons? Then, you used the source domain data to test the target domain. Have you tested how this compares with the target domain data test the target domain?

**Comment by Yu-Fangxu** [comment]
Thanks for your letter.
Sorry, I could not give you suggestions since I'm not quite familiar with the recommendation system. This paper is related to domain adaptation, so if you want to test the result in the same domain, there's no necessity to use this model. Maybe AutoRec and other VAE-based models may help.

> hello, thanks for your open code. In my comparative experiment, I changed the evaluation criterion from RMASE to top-N recommendation. We selected the largest N items as recommendations according to the predicted values, and used HR and NDCG as measurement standards. However, I found that the effect was a little poor, and if you have tried this evaluation method. Or some possible reasons? Then, you used the source domain data to test the target domain. Have you tested how this compares with the target domain data test the target domain?

**Comment by sunshinelium** [comment]
thanks for your replay, i want to compair it in cross-domain recommendation(the area your paper focuses on).  There is training data in both areas, and i just wonder why you test target domain performance by source train data. And i change the loss function in pretrain model and your model to cross-entropy like multvae, i found the performance better. But it was still a bit worse than the pre-trained AutoRec model even with two domains of pre-training applied.

**Comment by Yu-Fangxu** [comment]
Thanks.
Actually, I'm not the author of this paper. Many details are not clear such as their data preprocessing, datasets. Their goal is to let the model absorb knowledge in the source domain and exploit this knowledge to other domains. It is probably just a domain adaptation or transfer learning task. 
I have updated the source code the author sent to me. It would be great if it can help.

**Comment by sunshinelium** [comment]
Thanks for your reply. After I adjusted the regularIzation ratio, it had a more normal results.


 
---Original---
From: ***@***.***&gt;
Date: 2021/12/23 17:04:54
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [Yu-Fangxu/DARec] Some questions when reproducing code (Issue #1)




 
Thanks.
 Actually, I'm not the author of this paper. Many details are not clear such as their data preprocessing, datasets. Their goal is to let the model absorb knowledge in the source domain and exploit this knowledge to other domains. It is probably just a domain adaptation or transfer learning task.
 I have updated the source code the author sent to me. It would be great if it can help.
 
—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.
